plugins/uefi/fu-uefi-pcrs.c

```diff
@@ -103,10 +96,52 @@ fu_uefi_pcrs_setup_tpm12 (FuUefiPcrs *self, const gchar *fn_pcrs, GError **error
 }
 
 static gboolean
-fu_uefi_pcrs_setup_tpm20 (FuUefiPcrs *self, const gchar *argv0, GError **error)
+fu_uefi_pcrs_setup_tpm20 (FuUefiPcrs *self, GError **error)
 {
-	const gchar *argv[] = { argv0, NULL };
-	return fu_common_spawn_sync (argv, fu_uefi_pcrs_parse_line, self, 1500, NULL, error);
+	TSS2_RC rc;
+	ESYS_CONTEXT *ctx = NULL;
+	TPML_PCR_SELECTION pcr_selection_in = { 0, };
+	TPML_DIGEST *pcr_values = NULL;
```

Comment:
Would it be possible to do a custom `g_autoptr` for `TPML_DIGEST` and `ESYS_CONTEXT` to make cleanup and error handling better in this function?  More information about this stuff here: https://developer.gnome.org/glib/stable/glib-Miscellaneous-Macros.html#G-DEFINE-AUTOPTR-CLEANUP-FUNC:CAPS

Reply:
Done, added `G_DEFINE_AUTOPTR_CLEANUP_FUNC` for `ESYS_CONTEXT` and used `g_autofree` for `TPML_DIGEST`.